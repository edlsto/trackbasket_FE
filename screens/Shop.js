/* eslint-disable no-shadow */
import React, { useState, useContext } from 'react';
import { StyleSheet, Text, View, TextInput, ScrollView } from 'react-native';
import GroceryItem from '../components/GroceryItem';
import UserContext from '../user-context';

const sampleData = {
  id: 'searchItems',
  attributes: {
    items: [
      {
        upc: '0001111042852',
        aisleNumber: 7,
        description: 'Simple Truth Organicâ„¢ 2% Reduced Fat Milk',
        image_url:
          'https://silk.com/wp-content/uploads/2019/02/unsweet-almond-coconut-blend-1.png',
        price: 3.29,
        quantity: 1,
      },
      {
        upc: '8305729934',
        aisleNumber: 3,
        description: 'Nutella',
        image_url:
          'https://silk.com/wp-content/uploads/2019/02/unsweet-almond-coconut-blend-1.png',
        price: 4.59,
        quantity: 1,
      },
    ],
  },
};

const Shop = () => {
  const [text, setText] = useState('');
  const { attributes } = sampleData;
  const { items } = attributes;
  const { removeFromCart, addToCart, cart } = useContext(UserContext);
  const toggleCartItem = (upc) => {
    let selectedItem = items.find((item) => item.upc === upc);
    addToCart(selectedItem);
    if (cart.find((item) => item.upc === upc)) {
      removeFromCart(selectedItem);
    }
  };
  return (
    <View style={styles.container}>
      <View style={styles.innercontainer}>
        <TextInput
          placeholder="Search Items..."
          style={styles.searchBar}
          onChangeText={(text) => setText(text)}
          onSubmitEditing={() => console.log(text)}
        />
        {items.map((item) => {
          return (
            <GroceryItem
              upc={item.upc}
              aisleNumber={item.aisleNumber}
              description={item.description}
              image_url={item.image_url}
              price={item.price}
              clickHandler={toggleCartItem}
              quantity={item.quantity}
              key={item.upc}
            />
          );
        })}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
  },
  innercontainer: {
    width: 350,
  },
  searchBar: {
    borderWidth: 1,
    fontSize: 18,
    height: 50,
    margin: 10,
    padding: 5,
  },
});

export default Shop;
