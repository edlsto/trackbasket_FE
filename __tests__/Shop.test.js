import React from 'react';
import Shop from '../screens/Shop';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<Shop />).toJSON();
  expect(tree).toMatchSnapshot();
});
