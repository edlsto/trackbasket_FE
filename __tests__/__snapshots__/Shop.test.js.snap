// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 350,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Search Items..."
      rejectResponderTermination={true}
      style={
        Object {
          "borderWidth": 1,
          "fontSize": 18,
          "height": 50,
          "marginBottom": 40,
          "marginTop": 10,
          "padding": 5,
        }
      }
      underlineColorAndroid="transparent"
    />
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "alignItems": "stretch",
              "marginBottom": 40,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderWidth": 1,
                  "marginRight": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://silk.com/wp-content/uploads/2019/02/unsweet-almond-coconut-blend-1.png",
                  }
                }
                style={
                  Object {
                    "height": 100,
                    "marginBottom": 15,
                    "marginTop": 15,
                    "width": 110,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                Simple Truth Organicâ„¢ 2% Reduced Fat Milk
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                $
                3.29
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginHorizontal": 20,
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderWidth": 1,
                    "fontSize": 20,
                    "marginHorizontal": 10,
                    "textAlign": "center",
                    "width": 40,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <Text />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "lightgray",
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Helvetica",
                    "fontSize": 18,
                  }
                }
              >
                Add to cart
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "stretch",
              "marginBottom": 40,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderWidth": 1,
                  "marginRight": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://user-images.githubusercontent.com/4350550/83094330-b3ac3980-a05e-11ea-97fb-9dfb29bc817b.png",
                  }
                }
                style={
                  Object {
                    "height": 100,
                    "marginBottom": 15,
                    "marginTop": 15,
                    "width": 110,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                Nature's Own Honey Wheat Sliced Bread
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                $
                2.99
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginHorizontal": 20,
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderWidth": 1,
                    "fontSize": 20,
                    "marginHorizontal": 10,
                    "textAlign": "center",
                    "width": 40,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <Text />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "lightgray",
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Helvetica",
                    "fontSize": 18,
                  }
                }
              >
                Add to cart
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "stretch",
              "marginBottom": 40,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderWidth": 1,
                  "marginRight": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://user-images.githubusercontent.com/4350550/83094526-17cefd80-a05f-11ea-9856-7a5c8c3566e5.png",
                  }
                }
                style={
                  Object {
                    "height": 100,
                    "marginBottom": 15,
                    "marginTop": 15,
                    "width": 110,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                Klondike The Original Ice Cream Bars
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                $
                3.49
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginHorizontal": 20,
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderWidth": 1,
                    "fontSize": 20,
                    "marginHorizontal": 10,
                    "textAlign": "center",
                    "width": 40,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <Text />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "lightgray",
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Helvetica",
                    "fontSize": 18,
                  }
                }
              >
                Add to cart
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "stretch",
              "marginBottom": 40,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderWidth": 1,
                  "marginRight": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://user-images.githubusercontent.com/4350550/83095102-dc80fe80-a05f-11ea-9027-73ae65963359.png",
                  }
                }
                style={
                  Object {
                    "height": 100,
                    "marginBottom": 15,
                    "marginTop": 15,
                    "width": 110,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                KrogerÂ® Restaurant Style Tortilla Chips
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                $
                1.25
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginHorizontal": 20,
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderWidth": 1,
                    "fontSize": 20,
                    "marginHorizontal": 10,
                    "textAlign": "center",
                    "width": 40,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <Text />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "lightgray",
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Helvetica",
                    "fontSize": 18,
                  }
                }
              >
                Add to cart
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "stretch",
              "marginBottom": 40,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "white",
                  "borderWidth": 1,
                  "marginRight": 20,
                }
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://user-images.githubusercontent.com/4350550/83095372-6fba3400-a060-11ea-91fc-f646038c2dfd.png",
                  }
                }
                style={
                  Object {
                    "height": 100,
                    "marginBottom": 15,
                    "marginTop": 15,
                    "width": 110,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                Simple Truth Organicâ„¢ Gala Apples Pouch
              </Text>
              <Text
                style={
                  Object {
                    "fontSize": 22,
                  }
                }
              >
                $
                3.99
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginHorizontal": 20,
                "marginTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                rejectResponderTermination={true}
                style={
                  Object {
                    "borderWidth": 1,
                    "fontSize": 20,
                    "marginHorizontal": 10,
                    "textAlign": "center",
                    "width": 40,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
              <Text />
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "lightgray",
                  "opacity": 1,
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "Helvetica",
                    "fontSize": 18,
                  }
                }
              >
                Add to cart
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
